### PR TITLE
Unified graphs

### PR DESCRIPTION
**Pull request by chatman**
Currently, the open source graphs (http://mostly.cool) are using a different generation logic (based on createBranchGraphs.py). Intention here is to port those graphs over to the graph-scripts/ logic. However, there are a few gaps in functionality there, which I'm trying to address:

1. We need support for plotting multiple tests at the same time (across multiple branches).
2. A single test run results in multiple graphs (one for each task). We should be able to combine them into a single graph. Motivation for this usecase is that going forward, there might be tens/hundreds of tests added each with multiple tasks, so having one graph per test (for a given branch) would make it more readable when scanning through for regressions.

Here's the first step towards that:
* Support multiple tests in a graph. Here, the -r parameter can now take multiple result dirs.

### CODE REVIEW

**Comment by chatman** [comment]
Here's an example of two graphs with this:
[graph.zip](https://github.com/fullstorydev/solr-bench/files/10776437/graph.zip)

**Comment by chatman** [comment]
I've published the graphs using the latest commit to http://mostly.cool/graph.html.

Click "branch_9x", then "Collapse" button on top. Here's how it looks for me:
image

**Comment by chatman** [comment]
TODO: bring back the task descriptions, possibly even link the suites on GH.

**Comment by chatman** [comment]
@patsonluk When you please get a chance, can you weigh in on this idea (collapsed vs. expanded)?

**Comment by justinrsweeney** [comment]
My two cents is that we should default to collapsed. I think it provides a more concise overall view and users can expand for additional details.

File: README.md [hunk 1]
```diff
@@ -51,49 +51,39 @@ Example: `./cleanup.sh ./stress.sh -c dfde16a004206cc92e21cc5a6cad9030fbe13c20 s
...
 ```
 
 1. `cluster-test.json` : Creates an 8 node cluster and create 1000 collections of various `numShards` and measure shutdown & restart performance
 2. `stress-facets-local.json` : Indexes 20 million documents from an ecommerce events dataset, issues 5k facet queries against it.
+3. `prs-vs-nonprs.json` : Creates 1k collections using non-PRS mode (default), then restarts 7 nodes, then cleans up, and repeats the same for PRS.
 
 ### Results
 
-* Results are available after the benchmark in `./suites/results/results-\<configfile\>-\<commit-id\>.json` file. 
+* Results are available after the benchmark in `./suites/results/<testname (without the .json suffix)>/<commit-id>/results.json` file.
```
Comment (patsonluk) [inline README.md:63]: Thanks for updating this! I made the change in stress.sh but didn't update README.md 😓

**Comment by patsonluk** [comment]
@chatman Thanks for the proposal! Just want to ensure that I understand the motivation correctly:
>We need support for plotting multiple tests at the same time (across multiple branches).
So for example someone can have a test suite that does only queries, and another test suite only indexing. And we want to see all the results within the same report (ie html file)?

>A single test run results in multiple graphs (one for each task). 
This is the current implementation before this proposed change as I understand it :) I actually broke it up this way as in our scenario, having all tasks within the same graph (restart/index/query) didn't make too much sense  

>We should be able to combine them into a single graph. Motivation for this usecase is that going forward, there might be tens/hundreds of tests added each with multiple tasks, so having one graph per test (for a given branch) would make it more readable when scanning through for regressions.
I assume `Per test` refers to test template/suite. not the test runs themselves? So displaying multiple tasks within the same test suite on a single graph? I assume the motivation is to lower the number of graphs generated (vs one graph per task) 😊 .

If that's the case, then this might make sense if all the tasks have similar durations. Otherwise having tasks with median of much longer duration might hide anomaly of tasks with short durations.

**Comment by chatman** [comment]
Sorry for missing your comment here! :-( Hence the late reply.

> So for example someone can have a test suite that does only queries, and another test suite only indexing. And we want to see all the results within the same report (ie html file)?

Yes

> This is the current implementation before this proposed change as I understand it :) I actually broke it up this way as in our scenario, having all tasks within the same graph (restart/index/query) didn't make too much sense - as restart has very high duration with makes the line for querying (median per query) almost invisible.

Yes, that makes sense. Stuffing all the different tasks into a single graph (esp with different units at times) can be confusing, and separate graphs per task makes sense. However, it is good in cases where the numbers of test suites themselves are few. In open source graphs (mostly.cool/graph.html at the moment), there will conceivably be 100s of test suites (testing various parts of Solr's functionality), hence introduced this "collapsed" mode.

> I assume Per test refers to test template/suite. not the test runs themselves? So displaying multiple tasks within the same test suite on a single graph? I assume the motivation is to lower the number of graphs generated (vs one graph per task) blush .

Right, "per test" == "per test suite". Exactly, you're right about the motivation.

> If that's the case, then this might make sense if all the tasks have similar durations. Otherwise having tasks with median of much longer duration might hide anomaly of tasks with short durations.

Yes, you're right. To mitigate that (surface the anamolies of shorter tasks better), I introduced the click to hide/show tasks in the legend. Not ideal, but hopefully gets the job done manually by hiding the longer duration tasks and only showing the shorter ones, when inspecting the graphs.

**Comment by chatman** [comment]
> My two cents is that we should default to collapsed. I think it provides a more concise overall view and users can expand for additional details.

I've implemented this. All single group tabs now have collapsed graphs by default, and the "Expand" button works by showing graphs on a per task basis, as Patson implemented them. WDYT, @patsonluk . In case you have an objection to making this default, I can work on adding some configuration parameter for collapsed vs expanded by default to the python based generation script, and pass it through to the JS.

**Comment by chatman** [comment]
Merging this for progress. One thing we still need to do (perhaps in a separate ticket) is to bring back the descriptions of the tasks from the config files to be shown on the graphs.

**Comment by patsonluk** [comment]
> Merging this for progress. One thing we still need to do (perhaps in a separate ticket) is to bring back the descriptions of the tasks from the config files to be shown on the graphs.

Thanks @chatman ! I will merge my PR as well and try it out!

**Comment by chatman** [comment]
Sounds perfect. Thanks!

**Pull request merged**